src/Microsoft.TestPlatform.TestHostProvider/Hosting/DefaultTestHostManager.cs

```diff
@@ -259,10 +261,10 @@ private int LaunchHost(TestProcessStartInfo testHostStartInfo)
             catch (OperationCanceledException ex)
             {
                 this.OnHostExited(new HostProviderEventArgs(ex.Message, -1, 0));
-                return -1;
+                throw;
```

Comment:
Yes, there is no reason that LaunchTest should throw, the caller can as much get the information from the exited event.